### PR TITLE
Added correct return type for stdin and parameterized stdin source

### PR DESCRIPTION
**Pull request by Jegp**
Corrected the return type of the `StdEventInput` module and improved stdio arguments and encodings.
I also improved the CLI help text a bit.

### CODE REVIEW

File: faery/stdio.py [hunk 1]
```diff
@@ -18,15 +24,54 @@ def apply(self, data: Events):
...
+            self.file.close()
+
+    def _vacate_buffer(self):
+        array = np.array(self.buffer, dtype=Event)
+        self.buffer = []
+        return array
+
+    def __next__(self) -> Events:
+        while len(self.buffer) <= self.BUFFER_SIZE:
+            line = self.file.readline()
```
[the remark this answers is not in the document]
Comment (andreapignaz) [reply faery/stdio.py:69]: In the last commit, the use of `strip()` solves the problem.

File: faery/csv.py [hunk 2]
```diff
@@ -15,27 +15,34 @@ class CsvEventStreamIterator(StreamIterator):
 
     CSV_PATTERN = re.compile(r"(\d*),(\d*),(\d*),(\d)")
 
+    buffer: List[Tuple[int, int, int, bool]]
```
Comment (andreapignaz) [inline faery/csv.py:18]: `buffer: List[Events]` generates an Exception when importing the library, as Events is not a standard type but a Numpy datatype. This solves the problem, like it's done in the `StdEventInputIterator` class.

**Review by andreapignaz — COMMENTED** [review]
With the last commit I dealt with some problems that prevented `faery` from working correctly on my machine.

File: faery/csv.py (hunk 2 above)
Comment (Jegp) [reply faery/csv.py:18]: Great catch! Interesting about the exception, shouldn't that be reproduced in the GitHub actions test? I noticed all the checks were passing. Is there something we could/should add as a test to catch errors like this?

File: faery/stdio.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jegp) [reply faery/stdio.py:69]: Again, very very nice catch. I like the solution for now. We can perhaps add more testing in the future to catch these edge cases?
A separate concern that I discussed with Gregory from WSU is that we may want to pipe binary data at some point., I'll add that as an issue we can look into in the future...

**Review by Jegp — COMMENTED** [review]
Thank you for your feedback and code, Andrea! Very valuable and solid changes.
I'm curious whether we can somehow add a unit test that ensures that the library doesn't raise an exception on import. That's a nasty situation we really want to avoid. For now, I suggest merging the thing and keeping the discussion going on Discord.
